charts/capacity-prometheus-exporter/templates/deployment.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "capacity-prometheus-exporter.fullname" . }}
+  labels:
+    {{- include "capacity-prometheus-exporter.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "capacity-prometheus-exporter.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "capacity-prometheus-exporter.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "capacity-prometheus-exporter.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: metrics
+              containerPort: 8080
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /favicon.ico
```

Comment:
weird URL, why? Can we come up with a better one?

Reply:
The prometheus_client HTTP server answer with the collector function for any PATH it's called.

As this triggers the metrics gathering process that is relatively slow (~10s) and put load on the K8s metrics api, it's not a good idea to have the health check probes to it.

The exception is /favicon.ico PATH as this answer an empty 200OK. This is defined in the prometheus_client `make_wsgi_app` [here](https://github.com/prometheus/client_python/blob/master/prometheus_client/exposition.py#L121)